### DISCUSSION TITLE
"allow_scheduling_on_control_plane" disabled if "using_klipper_lb" is enabled.

### DISCUSSION THREAD

**Discussion by tijnhendrikson** [mysticaltech/terraform-hcloud-kube-hetzner#201] [Q&A] [answered]
Hi,

Thanks for the effort you've put into this project! 

I have a question regarding the "using_klipper_lb" option. I don't fully understand what the relation is between the "using_klipper_lb" option and to allow scheduling on control_plane.

The behavior seems unexpected to me, I was in the impression this was only controlled by the "allow_scheduling_on_control_plane". Is there a specific reason behind it?

I'm referring to this part:
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/b008bcf89a442447dc7f1329e56adc5b6ab11564/locals.tf#L82

Cheers!

**Comment by mysticaltech** [comment]
Thanks, @tijnhendrikson! Very good find... It is not correct indeed. What happened is that initially, we only activated Klipper for single-node clusters, so we had to allow scheduling on the control plane for obvious reasons, but then I added the option of using Klipper even for multi-node clusters, and the old logic stayed. 

So in line 82 above, we just need to replace `local.using_killer_lb` with `local.is_single_node_cluster`; please do not hesitate to submit a PR; I will merge it immediately. Otherwise, I will do it as soon as I can.

**Comment by mysticaltech** [comment, marked as answer]
@tijnhendrikson Did the fix! And released in v1.2.3. Thanks again!

**Reply by tijnhendrikson** [reply]
No way, you're too quick! Thanks

**Reply by schlichtanders** [reply]
I would like to turn off scheduling on control plane but still use klipper lb (or no extra load balancer) - just like the title of this discussion states.

The current implementation makes this impossible indeed, always if klipper_lb is active, scheduling is allowed on the control plane notes:
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/209d48a1a56e8e8e380e8aec67a64f3d588c2621/locals.tf#L109

so the discussion is not solved @tijnhendrikson can you unmark the answer so that this discussion is again open?

**Reply by schlichtanders** [reply]
@mysticaltech you seem to be the one who selected the answer - can you please unselect it, as this discussion is NOT solved
